src/core/thread/network_data_leader_ftd.cpp

```diff
@@ -1378,6 +1378,45 @@ Error Leader::RemoveStaleChildEntries(Coap::ResponseHandler aHandler, void *aCon
     return error;
 }
 
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_BORDER_ROUTING_ENABLE
+bool Leader::ContainsOmrPrefix(const Ip6::Prefix &aPrefix)
+{
+    PrefixTlv *prefixTlv;
+    bool       contains = false;
+
+    VerifyOrExit(BorderRouter::RoutingManager::IsValidOmrPrefix(aPrefix));
+
+    prefixTlv = FindPrefix(aPrefix);
+    VerifyOrExit(prefixTlv != nullptr);
+
+    for (int i = 0; i < 2; i++)
```

Comment:
This looks weird. Can we make line 1396~1411 a method?

Reply:
We have other places that follow the same style. 

We can create a iterator for that, but maybe i a future PR.

@abtink your thoughts?

Reply:
+1 on making it a method for readability

Reply:
It should be refactored in a separate PR since there are multiple similar usages